activerecord/test/cases/associations/has_and_belongs_to_many_associations_test.rb

```diff
@@ -95,6 +95,15 @@ def category
   has_and_belongs_to_many :projects, extend: NamedExtension
 end
 
+class ProjectUnscopingDavidDefaultScope < ActiveRecord::Base
+  self.table_name = 'projects'
+  has_and_belongs_to_many :developers, -> { unscope(where: 'name') },
+    :class_name => "LazyBlockDeveloperCalledDavid",
+    :join_table => "developers_projects",
+    :foreign_key => "project_id",
+    :association_foreign_key => "developer_id"
```

Comment:
New code added to Rails should use the new hash syntax. `foreign_key: "project_id"`